PowerShellTools/DebugEngine/ScriptDebugger.cs

```diff
@@ -219,7 +236,11 @@ public void UpdateBreakpoint(DebuggerBreakpointUpdatedEventArgs e)
         /// </summary>
         public void DebuggerFinished()
         {
-            IsDebuggingCommandReady = false;
+            lock (_debuggingCmdLock)
+            {
+                IsDebuggingCommandReady = false;

```

Comment:
But does the lock do anything here?  we don't read/use any values within these lock blocks, as far as I can tell, the only thing contained within them is setting IsDebuggingCommandReady .  Eg, one thread can set it to true; and another could set it to false right after, when we're off of the critical path...